tests/status/base/test.sh

```diff
@@ -77,10 +77,21 @@ rlJournalStart
         rlRun -s "tmt status -v"
         rlAssertGrep "$tmprun" $rlRun_LOG
         rlRun -s "tmt status --workdir-root=/var/tmp/tmt -v"
-        rlAssertGrep "/var/tmp/tmt" $rlRun_LOG
+        # If /var/tmp/tmt is empty - because user does not really use it - then
+        # the best we can do is to make sure the $tmprun does not appear in the
+        # output.
+        if [ ls /var/tmp/tmt &> /dev/null ]; then
```

Comment:
Hi @happz, looks `[ ls /var/tmp/tmt &> /dev/null ]` doesn't work. e.g.

```bash
huanli@ThinkPadMTV:tmp$ cat /tmp/foo.sh
#!/bin/bash

if [ ls /var/tmp/tmt &> /dev/null ]; then
    echo "GOOD: /var/tmp/tmt is NOT empty"
else
    echo "Oops: /var/tmp/tmt is empty"
    ls /var/tmp/tmt
fi

ls /var/tmp/tmt &> /dev/null; echo $?
[ ls /var/tmp/tmt &> /dev/null ]; echo $?
huanli@ThinkPadMTV:tmp$ 
huanli@ThinkPadMTV:tmp$ 
huanli@ThinkPadMTV:tmp$ # 1 - rm -rf /var/tmp/tmt
huanli@ThinkPadMTV:tmp$ rm -rf /var/tmp/tmt
huanli@ThinkPadMTV:tmp$ bash /tmp/foo.sh
Oops: /var/tmp/tmt is empty
ls: cannot access '/var/tmp/tmt': No such file or directory
2
2
huanli@ThinkPadMTV:tmp$ # 2 - mkdir /var/tmp/tmt
huanli@ThinkPadMTV:tmp$ mkdir /var/tmp/tmt
huanli@ThinkPadMTV:tmp$ bash /tmp/foo.sh
Oops: /var/tmp/tmt is empty
0
2
huanli@ThinkPadMTV:tmp$ # 3 - create a file and a subdir under /var/tmp/tmt
ThinkPadMTV:tmp$ touch /var/tmp/tmt/log1 && mkdir /var/tmp/tmt/run1
huanli@ThinkPadMTV:tmp$ bash /tmp/foo.sh
Oops: /var/tmp/tmt is empty
log1  run1
0
2
```

If `/var/tmp/tmt` does exist, `[ ls /var/tmp/tmt &> /dev/null ]` always returns `!0` (i.e. `2`). So, looks we should use ` if ls /var/tmp/tmt &> /dev/null; then` instead, e.g.

```bash
huanli@ThinkPadMTV:tmp$ cat /var/tmp/bar.sh 
#!/bin/bash

if ls /var/tmp/tmt &> /dev/null; then
    echo "GOOD: /var/tmp/tmt is NOT empty"
else
    echo "Oops: /var/tmp/tmt is empty"
    ls /var/tmp/tmt
fi

ls /var/tmp/tmt &> /dev/null; echo $?
huanli@ThinkPadMTV:tmp$ rm -rf /var/tmp/tmt && bash /tmp/bar.sh
Oops: /var/tmp/tmt is empty
ls: cannot access '/var/tmp/tmt': No such file or directory
2
huanli@ThinkPadMTV:tmp$ mkdir /var/tmp/tmt && bash /tmp/bar.sh
GOOD: /var/tmp/tmt is NOT empty
0
huanli@ThinkPadMTV:tmp$ touch /var/tmp/tmt/log1 && mkdir /var/tmp/tmt/run1 && bash /tmp/bar.sh
GOOD: /var/tmp/tmt is NOT empty
0
```

What do you think? Please correct me if what I understand is incorrect, thanks!

Reply:
Nicely spotted, it is indeed broken.

Eventually, I think the best would be `if ls /var/tmp/tmt/*` - note the wildcard `*` at the end:

```shell
#!/bin/bash

function check () {
  ls -al /var/tmp/tmt
  echo

  if ls /var/tmp/tmt &> /dev/null; then
    echo "ls /var/tmp/tmt &> /dev/null:         positive exit"
  else
    echo "ls /var/tmp/tmt &> /dev/null:         negative exit"
  fi

  if ls /var/tmp/tmt/* &> /dev/null; then
    echo "ls /var/tmp/tmt/* &> /dev/null ]:     positive exit"
  else
    echo "ls /var/tmp/tmt/* &> /dev/null ]:     negative exit"
  fi
}

echo "1. directory exists and contains a file"
mkdir -p /var/tmp/tmt
touch /var/tmp/tmt/foo

echo

check

echo

echo "2. directory exists and it's empty"
rm /var/tmp/tmt/foo

echo
check
echo

echo "3. directory does not exist"
rm -rf /var/tmp/tmt

echo
check
echo
```

```console
1. directory exists and contains a file

total 16
drwxr-xr-x.   2 root root  4096 Jun 20 11:02 .
drwxrwxrwt. 117 root root 12288 Jun 20 11:02 ..
-rw-r--r--.   1 root root     0 Jun 20 11:02 foo

ls /var/tmp/tmt &> /dev/null:         positive exit
ls /var/tmp/tmt/* &> /dev/null ]:     positive exit

2. directory exists and it's empty

total 16
drwxr-xr-x.   2 root root  4096 Jun 20 11:02 .
drwxrwxrwt. 117 root root 12288 Jun 20 11:02 ..

ls /var/tmp/tmt &> /dev/null:         positive exit
ls /var/tmp/tmt/* &> /dev/null ]:     negative exit

3. directory does not exist

ls: cannot access '/var/tmp/tmt': No such file or directory

ls /var/tmp/tmt &> /dev/null:         negative exit
ls /var/tmp/tmt/* &> /dev/null ]:     negative exit
```

What do you think?

Reply:
Yes, It is more precise to use `ls /var/tmp/tmt/*` instead of `ls /var/tmp/tmt`!